### ISSUE TITLE
Support `textlint-expect-error`

### ISSUE THREAD

**Issue by 3w36zj6** [textlint/textlint-filter-rule-comments#13]
Currently, `textlint-disable` does not report anything if no errors are present within the range where the rule is disabled. It would be helpful to have a feature similar to TypeScript's `@ts-expect-error`[^1], which reports if there are no errors within the disabled range.

[^1]: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments

**Comment by azu** [comment]
Can you give me an example of how it would be used?
Is it similar to ESLint's `--report-unused-disable-directives`?
https://eslint.org/docs/latest/use/command-line-interface#--report-unused-disable-directives

I think we'll probably need a new API or an option for `shouldIgnore` in textlint's context object.

📝 At the implementation level, it appears there are two options: ESLint's `--report-unused-disable-directives` approach or TypeScript's `@ts-expect-error` approach.

ESLint's `--report-unused-disable-directives` feels a bit cumbersome to handle since it adds another flag (there's also the issue that users won't use it unless it's the default).
I support the `@ts-expect-error` approach, but I think we need to give some thought to how the reporting format should be handled.

**Comment by 3w36zj6** [comment]
I wasn't aware of ESLint's `--report-unused-disable-directives` option.

Here is the scenario I have in mind for this use case:

First, suppose there is a false positive in the implementation of a rule, so a paragraph of "Lorem ipsum" incorrectly causes an error. The user partially disables the rule for this section.

```md
<!-- textlint-expect-error foo -->

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

<!-- textlint-enable -->
```

Next, when the rule version is updated, this false positive gets resolved. That is, the "Lorem ipsum" paragraph no longer reports an error on the new version, and the `textlint-expect-error` is now applied to a range with no errors.

In this case, an error should be reported for the `textlint-expect-error` comment that specifies a range where no errors occur.

For reference, Biome's `biome-ignore` behaves similarly to `@ts-expect-error`[^1].

[^1]: https://biomejs.dev/analyzer/suppressions/

**Comment by 3w36zj6** [comment]
I think the idea of expecting an error in the next node is a good one.

Additionally, it might also be useful to have `textlint-expect-error-start`, `textlint-expect-error-end`, and `textlint-expect-error-all` to expect errors within a specified range or for all nodes. This idea is similar to Biome's `biome-ignore-start`, `biome-ignore-end`, and `biome-ignore-all`.

**Comment by azu** [comment]
Start and end are probably excessive, so I think it's better to start with 'textlint-expect-error'
